goaviatrix/fqdn.go

```diff
@@ -483,3 +488,74 @@ func (c *Client) GetGwFilterTagList(fqdn *FQDN) (*FQDN, error) {
 	fqdn.GwFilterTagList = gwFilterTagList
 	return fqdn, nil
 }
+
+func (c *Client) AddFQDNDomainName(fqdn *FQDN) error {
+	action := "add_fqdn_filter_tag_domain_names"
+
+	policies, err := json.Marshal(fqdn.DomainList)
+	if err != nil {
+		return fmt.Errorf("could not marshal fqdn domain: %v", err)
+	}
+
+	return c.PostAPI(action, struct {
+		Action   string `form:"action"`
+		CID      string `form:"CID"`
+		TagName  string `form:"tag_name"`
+		Policies string `form:"policies"`
+	}{
+		Action:   action,
+		CID:      c.CID,
+		TagName:  fqdn.FQDNTag,
+		Policies: string(policies),
+	}, BasicCheck)
+}
+
+func (c *Client) GetFQDNDomainName(fqdn *FQDN) (*FQDN, error) {
+	foundFQDN, err := c.ListDomains(&FQDN{FQDNTag: fqdn.FQDNTag})
+	if err != nil {
+		return nil, fmt.Errorf("could not list fqdn domains: %v", err)
+	}
+	domain := fqdn.DomainList[0]
+	found := false
+	for _, d := range foundFQDN.DomainList {
```

Comment:
Same. Better to have an API to do this processing

Reply:
Opened mantis [16320](https://mantis.aviatrix.com/mantisbt/view.php?id=16320) proposing the new APIs for controller to do this processing.

Reply:
Sounds good. We can keep the current code till the API is available